docs/docs/reference/rendering-options/server-side-rendering.md

```diff
@@ -0,0 +1,133 @@
+---
+title: Server-side Rendering API
+---
+
+> **Note:** this feature requires running NodeJS server.
+> It is currently fully supported with [`gatsby serve`](/docs/reference/gatsby-cli/#serve) and in [Gatsby Cloud](/products/cloud/).
+
+Server-side Rendering (SSR) allows you to render a page at run-time with data that is fetched when a user visits the page.
+The server generates the full HTML during HTTP request and sends it to the user. The API is focused on data fetching outside of the Gatsby data-layer.
+
+## Creating server-rendered pages
+
+You can create server-rendered pages [the same way as regular pages](/docs/reference/routing/creating-routes/) (including using the File System Route API).
+
+The main difference is that page component must export an async function called `getServerData`:
+
+```js
+export async function getServerData(context) {
+  return {
+    props: {}, // Will be passed to the page component as "serverData" prop
+    headers: {}, // HTTP response headers for this page
+  }
+}
+```
+
+When a page component exports `getServerData` function, Gatsby treats all pages built with this component
+as server-rendered.
+
+The `context` parameter is an object with the following keys:
+
+- `headers`: The request headers
+- `method`: The request method, e.g. `GET`
+- `url`: The request URL
+- `query`: An object representing the query string
+- `params`: If you use [File System Route API](/docs/reference/routing/file-system-route-api/) the URL path gets passed in as `params`. For example, if your page is at `src/pages/{Product.name}.js` the `params` will be an object like `{ name: 'value' }`.
+
+`getServerData` can return an object with two keys:
+
+- `props` (optional): Object containing the data passed to `serverData` page prop. Should be a serializable object.
```

Comment:
Is `props` optional or required? In the How-To Guide it sounds like it's required: https://github.com/gatsbyjs/gatsby/pull/33111/files#diff-fc423777cbc4b7aeaee799ab669ac47b31f6609d5ae1bc3b144b3a7034ac8380R48